integration/tests/consensus/inclusion_test.go

```diff
@@ -102,34 +102,117 @@ func (is *InclusionSuite) TearDownTest() {
 }
 
 func (is *InclusionSuite) TestCollectionGuaranteeIncluded() {
-
+	is.T().Logf("------ test started")
 	// fix the deadline for the test as a whole
 	deadline := time.Now().Add(20 * time.Second)
 
 	// generate a sentinel collection guarantee
 	sentinel := unittest.CollectionGuaranteeFixture()
 	sentinel.SignerIDs = []flow.Identifier{is.collID}
 	sentinel.ReferenceBlockID = is.net.Root().ID()
+	colID := sentinel.CollectionID
+
+	is.waitUntilSeenProposal(deadline)
+
+	is.T().Logf("seen a proposal")
+
+	conID := is.sendCollectionToConsensus(deadline, sentinel)
+
+	is.T().Logf("sent collection %x to consensus %v", colID, conID)
 
-	is.T().Logf("collection guarantee generated: %x\n", sentinel.CollectionID)
+	proposal := is.waitUntilCollectionIncludeInProposal(deadline, sentinel)
+
+	is.T().Logf("collection guarantee %x included in a proposal %x\n", colID, proposal.Header.ID())
+
+	is.waitUntilProposalConfirmed(deadline, sentinel, proposal)
+
+	is.T().Logf("collection guarantee %x is confirmed 3 times\n", colID)
+}
 
+func (is *InclusionSuite) waitUntilSeenProposal(deadline time.Time) {
+	for time.Now().Before(deadline) {
+
+		// we read the next message until we reach deadline
+		_, msg, err := is.reader.Next()
+		if err != nil {
+			is.T().Logf("could not read message: %s\n", err)
+			continue
+		}
+
+		// we only care about block proposals at the moment
+		_, ok := msg.(*messages.BlockProposal)
+		if !ok {
+			continue
+		}
+		return
+	}
+	is.T().Fatal("timeout waiting to see proposal")
+}
+
+func (is *InclusionSuite) sendCollectionToConsensus(deadline time.Time, sentinel *flow.CollectionGuarantee) flow.Identifier {
+	colID := sentinel.CollectionID
+
+	conID := is.conIDs[0]
 	// keep trying to send collection guarantee to at least one consensus node
-SendingLoop:
 	for time.Now().Before(deadline) {
-		conID := is.conIDs[rand.Intn(len(is.conIDs))]
 		ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+		is.T().Logf("sending collection %x to consensus node %v\n", colID, conID)
 		err := is.Collection().Send(ctx, engine.PushGuarantees, sentinel, conID)
 		cancel()
 		if err != nil {
 			is.T().Logf("could not send collection guarantee: %s\n", err)
 			continue
 		}
-		break SendingLoop
+
+		return conID
 	}
+	is.T().Fatalf("timeout sendng collection %x to consensus", colID)
+	return flow.ZeroID
+}
 
+func (is *InclusionSuite) waitUntilCollectionIncludeInProposal(deadline time.Time, sentinel *flow.CollectionGuarantee) *messages.BlockProposal {
+	colID := sentinel.CollectionID
+	// we try to find a block with the guarantee included and three confirmations
```

Comment:
```suggestion
	// we try to find a block with the guarantee included
```
We are checking confirmations/finalization later